<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>598</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Object Detect</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="1" column="1">
     <widget class="QProgressBar" name="progressBar">
      <property name="value">
       <number>0</number>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QScrollArea" name="scrollArea">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>350</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustIgnored</enum>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>332</width>
         <height>698</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QFrame" name="frame">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <property name="lineWidth">
           <number>0</number>
          </property>
          <layout class="QGridLayout">
           <item row="4" column="0">
            <widget class="QGroupBox" name="groupBox">
             <property name="title">
              <string>Information about last kernel</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_10">
              <item row="3" column="0">
               <widget class="QLabel" name="labelGoodMatches">
                <property name="text">
                 <string>Good Matches</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1" rowspan="4">
               <layout class="QGridLayout" name="gridLayout">
                <item row="0" column="1">
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>Object / Scene</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QLineEdit" name="lineEditMatches"/>
                </item>
                <item row="3" column="1">
                 <widget class="QLineEdit" name="lineEditGoodMatches"/>
                </item>
                <item row="1" column="1">
                 <widget class="QLineEdit" name="lineEditKeyPoints">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0" rowspan="4">
                 <spacer name="horizontalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>18</width>
                    <height>98</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="labelMatches">
                <property name="text">
                 <string>Matches</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="labelKeyPoints">
                <property name="text">
                 <string>KeyPoints</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QGroupBox" name="groupBox_2">
             <property name="title">
              <string>SURF Parameters</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="3" column="0">
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>Extended</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="lbl_oc">
                <property name="text">
                 <string>Number of octaves</string>
                </property>
               </widget>
              </item>
              <item row="5" column="0" colspan="2">
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>Filter Matches</string>
                </property>
               </widget>
              </item>
              <item row="5" column="2">
               <widget class="QSpinBox" name="spinBoxGoodMatchesThreshold">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>100</number>
                </property>
                <property name="value">
                 <number>20</number>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QSpinBox" name="spinBoxMinHessian">
                <property name="minimum">
                 <number>10</number>
                </property>
                <property name="maximum">
                 <number>800</number>
                </property>
                <property name="value">
                 <number>100</number>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>SURF Threshold</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Octave layers</string>
                </property>
               </widget>
              </item>
              <item row="3" column="2">
               <widget class="QCheckBox" name="checkBoxExtended">
                <property name="layoutDirection">
                 <enum>Qt::RightToLeft</enum>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="2" column="2">
               <widget class="QSpinBox" name="spinBox">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="value">
                 <number>2</number>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QSpinBox" name="spinBoxOctaves">
                <property name="value">
                 <number>4</number>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>Upright only</string>
                </property>
               </widget>
              </item>
              <item row="4" column="2">
               <widget class="QCheckBox" name="checkBoxUpright">
                <property name="layoutDirection">
                 <enum>Qt::RightToLeft</enum>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QGroupBox" name="groupBox_3">
             <property name="title">
              <string>Display options</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="0">
               <widget class="QCheckBox" name="checkBoxDrawKeyPoints">
                <property name="text">
                 <string>Draw Keypoints</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QCheckBox" name="checkBoxDrawMatches">
                <property name="text">
                 <string>Draw Matches</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QCheckBox" name="checkBoxLocaliseObject">
                <property name="text">
                 <string>Localise Object</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QGroupBox" name="groupBox_4">
             <property name="title">
              <string>Input</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_11">
              <item row="0" column="0">
               <widget class="QPushButton" name="pushButtonSourceImage">
                <property name="text">
                 <string>Object</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QPushButton" name="pushButtonSceneImage">
                <property name="text">
                 <string>Scene</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QGroupBox" name="groupBox_11">
             <property name="title">
              <string>Execution time in ms</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_9">
              <item row="0" column="0">
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>CPU</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1" rowspan="2">
               <layout class="QGridLayout" name="gridLayout_7">
                <item row="0" column="1">
                 <widget class="QLineEdit" name="lineEditExecutionTimeCpu">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QLineEdit" name="lineEditExecutionTimeGpu">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0" rowspan="2">
                 <spacer name="horizontalSpacer">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>58</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>GPU</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="6" column="0">
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
          <zorder>groupBox_2</zorder>
          <zorder>groupBox</zorder>
          <zorder>groupBox_4</zorder>
          <zorder>groupBox_3</zorder>
          <zorder>groupBox_11</zorder>
          <zorder>verticalSpacer</zorder>
         </widget>
        </item>
       </layout>
       <zorder>pushButtonSceneImage</zorder>
       <zorder>pushButtonSourceImage</zorder>
       <zorder>frame</zorder>
      </widget>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLabel" name="labelOutputImage">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>1</width>
        <height>1</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="text">
       <string>Processed Image</string>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="groupBox_5">
      <property name="title">
       <string>Execute</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="pushButtonCpu">
         <property name="text">
          <string>CPU</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButtonGpu">
         <property name="text">
          <string>GPU</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>pushButtonSourceImage</tabstop>
  <tabstop>pushButtonSceneImage</tabstop>
  <tabstop>checkBoxDrawMatches</tabstop>
  <tabstop>spinBoxMinHessian</tabstop>
  <tabstop>spinBoxGoodMatchesThreshold</tabstop>
  <tabstop>pushButtonCpu</tabstop>
  <tabstop>pushButtonGpu</tabstop>
  <tabstop>lineEditKeyPoints</tabstop>
  <tabstop>lineEditMatches</tabstop>
  <tabstop>lineEditGoodMatches</tabstop>
  <tabstop>lineEditExecutionTimeCpu</tabstop>
  <tabstop>lineEditExecutionTimeGpu</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
